File: scala-pr/54/scala/code-quality-error-handling/NumberDivider.scala
class NumberDivider {
  def divide(a: Double, b: Double): Double = {
    // {fact rule=code-quality-error-handling@v1.0 defects=1}
    try {
      if (b == 0) return 0.0 // Hiding division by zero
      a / b
    } catch {
      case _: Exception => Double.NaN
    }
    // {/fact}
  }
}

File: kotlin-pr/50/kotlin/code-quality-error-handling/NumberDivider.kt
class NumberDivider {
    fun divide(a: Double, b: Double): Double {
        // {fact rule=code-quality-error-handling@v1.0 defects=1}
        return try {
            if (b == 0.0) return 0.0 // Hiding division by zero
            a / b
        } catch (e: Exception) {
            Double.NaN
        }
        // {/fact}
    }
}